### PR TITLE
Mn/graphql

### PR DESCRIPTION
**Pull request by matydn05**
Setting Graphql up. There's going to be merge conflicts as the files were branched from an out of date version of mn/database, I'm going to try to see if I can get rid of them myself.

### CODE REVIEW

**Comment by matydn05** [comment]
I believe I fixed the conflicts.

File: src/schema/types/mutation.graphql [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # This is an empty type, meant to be extended by other entities.
 type Mutation {
-  _empty: String
+    createuser(username: String!,firstName: String!,lastName: String!,password: String!): User
```
Comment (mpautasso) [inline src/schema/types/mutation.graphql]: Let's leave a space after `,`

**Review by mpautasso — CHANGES_REQUESTED** [review]

**Comment by mpautasso** [comment]
@matydn05 Great, only a minor change.

File: src/schema/types/mutation.graphql (hunk 1 above)
Comment (abelosorio) [reply src/schema/types/mutation.graphql]: Mati, the idea of having different files is that you can compose mutations, types, and queries. So, the `mutation.graphql` would have only an `_empty` field (as query.graphql), and then you can create different files with more mutations, queries, and types. In this case, you could create a `user.graphql` file with everything you need for your user (fields, queries, and mutations):

user.graphql
```graphql
type User {
...
}

type Query {
  users: [User]
  user(...): User
}

type Mutation {
  createUser(...): User
}
```

These files are merged into a single file so, for example, the type `Mutation` will have all the mutations that you define in your files.

Let me know if I'm clear enough or we can hop on a call.

Comment (matydn05) [reply src/schema/types/mutation.graphql]: I think I got it, it's done.

File: src/schema/types/user.graphql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  lastName: String
+}
+
+type Query {
+  users: [User]!
+  user(id: ID!): User
+}
+
+type Mutation {
+    createuser(username: String!, firstName: String!, lastName: String!, password: String!): User
```
Comment (abelosorio) [inline src/schema/types/user.graphql]: I think you have 4-space indentation here. Please, replace it with 2.

Comment (matydn05) [reply src/schema/types/user.graphql]: Replaced.

**Review by abelosorio — APPROVED** [review]
:+1:

**Pull request merged**